lib/zhr_devs/identity_management/identity.ex

```diff
@@ -26,6 +26,19 @@ defmodule ZhrDevs.Identity do
     GenServer.start_link(__MODULE__, success_struct, name: via_tuple(hashed_identity))
   end
 
+  @spec renew_login(binary()) :: :ok | no_return()
+  def renew_login(hashed_identity) when is_binary(hashed_identity) do
+    GenServer.call(via_tuple(hashed_identity), :renew_login)
+  end
+
+  @spec digest_to_urlsafe!(binary()) :: Uptight.Result.t()
```

Comment:
Remember that we don't need to lose type information anymore! It was recently fixed by introducing `Uptight.Result.sum` type! So at least, write `Uptight.Result.sum(any, Uptight.Base.Urlsafe.t())` here, but better understand what sort of exception does `assert Result.is_ok?` throw (I assume, `MatchError`) and mention it explicitly: `Result.sum(MatchError.t(), Urlsafe.t())`.